Toggl.Giskard/Activities/LoginActivity.cs

```
@@ -32,15 +37,23 @@ protected override void OnCreate(Bundle bundle)
         private void setupToolbar()
         {
             var toolbar = FindViewById<Toolbar>(Resource.Id.LoginToolbar);
-
             toolbar.Title = ViewModel.Title;
 
             SetSupportActionBar(toolbar);
-
             SupportActionBar.SetDisplayHomeAsUpEnabled(true);
             SupportActionBar.SetDisplayShowHomeEnabled(true);
 
             toolbar.NavigationClick += navigationClick;
+            disposable = ViewModel.WeakSubscribe<PropertyChangedEventArgs>(nameof(ViewModel.Title), onTitleChanged);
+        }
+
+        private void onTitleChanged(object sender, PropertyChangedEventArgs args)
+        {
+            var toolbar = FindViewById<Toolbar>(Resource.Id.LoginToolbar);
+            if (toolbar != null)
+            {
+                toolbar.Title = ViewModel.Title;
```

I would not check for nulls here. Let the app crash so we know in development time that there's no toolbar in the xml.

---

Hmm, that's fair. I'll remove the null check then.